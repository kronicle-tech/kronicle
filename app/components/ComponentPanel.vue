<template>
  <div v-if="component">
    <h4><ComponentName :component="component" /></h4>

    <div v-if="component.teams && component.teams.length > 0">
      <h5>Teams</h5>
      <ComponentTeams :component-teams="component.teams" />
    </div>

    <div v-if="component.tags && component.tags.length > 0">
      <h5>Tags</h5>
      <Tags :tags="component.tags" />
    </div>

    <div v-if="component.description">
      <h5>Description</h5>
      <Markdown :markdown="component.description" />
    </div>

    <div v-if="component.notes">
      <h5>Notes</h5>
      <Markdown :markdown="component.notes" />
    </div>

    <div
      v-if="component.responsibilities && component.responsibilities.length > 0"
    >
      <h5>Responsibilities</h5>
      <Responsibilities :responsibilities="component.responsibilities" />
    </div>

    <div v-if="component.techDebts && component.techDebts.length > 0">
      <h5>Tech Debts</h5>
      <TechDebts :tech-debts="component.techDebts" />
    </div>

    <div v-if="component.openApiSpecs && component.openApiSpecs.length > 0">
      <h5>OpenAPI Specs</h5>
      <OpenApiSpecs :component="component" />
    </div>

    <div v-if="component.links && component.links.length > 0">
      <h5>Links</h5>
      <Links :links="component.links" />
    </div>
  </div>
</template>

<script lang="ts">
import Vue, { PropType } from 'vue'
import { Component } from '~/types/kronicle-service'
import ComponentName from '~/components/ComponentName.vue'
import Links from '~/components/Links.vue'
import Markdown from '~/components/Markdown.vue'
import OpenApiSpecs from '~/components/OpenApiSpecs.vue'
import ComponentTeams from '~/components/ComponentTeams.vue'
import Responsibilities from '~/components/Responsibilities.vue'
import Tags from '~/components/Tags.vue'
import TechDebts from '~/components/TechDebts.vue'

export default Vue.extend({
  components: {
    ComponentName,
    Links,
    Markdown,
    OpenApiSpecs,
    ComponentTeams,
    Responsibilities,
    Tags,
    TechDebts,
  },
  props: {
    component: {
      type: Object as PropType<Component>,
      default: undefined,
    },
  },
})
</script>
