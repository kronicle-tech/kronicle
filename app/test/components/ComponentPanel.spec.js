import { mount } from '@vue/test-utils'
import ComponentPanel from '@/components/ComponentPanel.vue'

describe('ComponentPanel', () => {
  let propsData
  let wrapper
  const createWrapper = () => {
    wrapper = mount(ComponentPanel, { propsData })
  }

  beforeEach(() => {
    propsData = {}
  })

  afterEach(() => {
    wrapper.destroy()
    wrapper = null
  })

  describe('when component prop is undefined', () => {
    test('renders nothing', () => {
      createWrapper()
      expect(wrapper.html()).toEqual('')
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('when component is set to a component', () => {
    beforeEach(() => {
      propsData.component = {
        id: 'test-id',
        name: 'Test Name',
      }
    })

    test("renders a h3 tag containing the component name and a link to the component's page", () => {
      createWrapper()
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('when component is set to just a component id', () => {
    beforeEach(() => {
      propsData.component = {
        id: 'test-id',
      }
    })

    test('renders a h3 tag containing the component id', () => {
      createWrapper()
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('when teams is set', () => {
    beforeEach(() => {
      propsData.component = {
        id: 'test-id',
        name: 'Test Name',
        teams: [
          {
            teamId: 'test-team-id-1',
          },
          {
            teamId: 'test-team-id-2',
          },
        ],
      }
    })

    test('renders the teams', () => {
      createWrapper()
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('when tags is set', () => {
    beforeEach(() => {
      propsData.component = {
        id: 'test-id',
        name: 'Test Name',
        tags: ['test-tag-1', 'test-tag-2'],
      }
    })

    test('renders the tags', () => {
      createWrapper()
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('when description is set', () => {
    beforeEach(() => {
      propsData.component = {
        id: 'test-id',
        name: 'Test Name',
        description: 'Description',
      }
    })

    test('renders the description', () => {
      createWrapper()
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('when notes is set', () => {
    beforeEach(() => {
      propsData.component = {
        id: 'test-id',
        name: 'Test Name',
        notes: 'Notes',
      }
    })

    test('renders the notes', () => {
      createWrapper()
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('when responsibilities is set', () => {
    beforeEach(() => {
      propsData.component = {
        id: 'test-id',
        name: 'Test Name',
        responsibilities: [
          {
            description: 'Test Description 1',
          },
          {
            description: 'Test Description 2',
          },
        ],
      }
    })

    test('renders the responsibilities', () => {
      createWrapper()
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('when links is set', () => {
    beforeEach(() => {
      propsData.component = {
        id: 'test-id',
        name: 'Test Name',
        links: [
          {
            url: 'https://example.org/test-1',
          },
          {
            url: 'https://example.org/test-2',
          },
        ],
      }
    })

    test('renders the links', () => {
      createWrapper()
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('when techDebts is set', () => {
    beforeEach(() => {
      propsData.component = {
        id: 'test-id',
        name: 'Test Name',
        techDebts: [
          {
            description: 'Test Description 1',
          },
          {
            description: 'Test Description 2',
          },
        ],
      }
    })

    test('renders the techDebts', () => {
      createWrapper()
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
