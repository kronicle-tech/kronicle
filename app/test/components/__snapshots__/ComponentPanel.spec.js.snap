// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentPanel when component is set to a component renders a h3 tag containing the component name and a link to the component's page 1`] = `
"<div>
  <h3 class=\\"text-info\\">Test Name</h3> <a href=\\"/components/test-id\\" target=\\"_self\\" class=\\"btn my-1 btn-info\\">
    More Info
  </a>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
</div>"
`;

exports[`ComponentPanel when component is set to just a component id renders a h3 tag containing the component id 1`] = `
"<div>
  <h3 class=\\"text-info\\">test-id</h3>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
</div>"
`;

exports[`ComponentPanel when component prop is undefined renders nothing 1`] = `""`;

exports[`ComponentPanel when description is set renders the description 1`] = `
"<div>
  <h3 class=\\"text-info\\">Test Name</h3> <a href=\\"/components/test-id\\" target=\\"_self\\" class=\\"btn my-1 btn-info\\">
    More Info
  </a>
  <div class=\\"my-4\\">
    <h5 class=\\"text-info\\">Description</h5>
    <div>
      <p>Description</p>
    </div>
  </div>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
</div>"
`;

exports[`ComponentPanel when links is set renders the links 1`] = `
"<div>
  <h3 class=\\"text-info\\">Test Name</h3> <a href=\\"/components/test-id\\" target=\\"_self\\" class=\\"btn my-1 btn-info\\">
    More Info
  </a>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <div class=\\"my-4\\">
    <h5 class=\\"text-info\\">Links</h5>
    <ul>
      <li><a href=\\"https://example.org/test-1\\" target=\\"_blank\\">https://example.org/test-1</a> <br>
        <!---->
      </li>
      <li><a href=\\"https://example.org/test-2\\" target=\\"_blank\\">https://example.org/test-2</a> <br>
        <!---->
      </li>
    </ul>
  </div>
  <!---->
</div>"
`;

exports[`ComponentPanel when notes is set renders the notes 1`] = `
"<div>
  <h3 class=\\"text-info\\">Test Name</h3> <a href=\\"/components/test-id\\" target=\\"_self\\" class=\\"btn my-1 btn-info\\">
    More Info
  </a>
  <!---->
  <div class=\\"my-4\\">
    <h5 class=\\"text-info\\">Notes</h5>
    <div>
      <p>Notes</p>
    </div>
  </div>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
</div>"
`;

exports[`ComponentPanel when responsibilities is set renders the responsibilities 1`] = `
"<div>
  <h3 class=\\"text-info\\">Test Name</h3> <a href=\\"/components/test-id\\" target=\\"_self\\" class=\\"btn my-1 btn-info\\">
    More Info
  </a>
  <!---->
  <!---->
  <div class=\\"my-4\\">
    <h5 class=\\"text-info\\">Responsibilities</h5>
    <ul>
      <li>
        <div>
          <p>Test Description 1</p>
        </div>
      </li>
      <li>
        <div>
          <p>Test Description 2</p>
        </div>
      </li>
    </ul>
  </div>
  <!---->
  <!---->
  <!---->
  <!---->
</div>"
`;

exports[`ComponentPanel when tags is set renders the tags 1`] = `
"<div>
  <h3 class=\\"text-info\\">Test Name</h3> <a href=\\"/components/test-id\\" target=\\"_self\\" class=\\"btn my-1 btn-info\\">
    More Info
  </a>
  <!---->
  <!---->
  <!---->
  <!---->
  <div class=\\"my-4\\">
    <h5 class=\\"text-info\\">Tags</h5>
    <ul>
      <li>
        <!---->
      </li>
      <li>
        <!---->
      </li>
    </ul>
  </div>
  <!---->
  <!---->
</div>"
`;

exports[`ComponentPanel when teams is set renders the teams 1`] = `
"<div>
  <h3 class=\\"text-info\\">Test Name</h3> <a href=\\"/components/test-id\\" target=\\"_self\\" class=\\"btn my-1 btn-info\\">
    More Info
  </a>
  <!---->
  <!---->
  <!---->
  <div class=\\"my-4\\">
    <h5 class=\\"text-info\\">Teams</h5>
    <ul>
      <li><a href=\\"/teams/test-team-id-1\\" class=\\"team-name\\">test-team-id-1</a>
        <!---->
        <!---->
      </li>
      <li><a href=\\"/teams/test-team-id-2\\" class=\\"team-name\\">test-team-id-2</a>
        <!---->
        <!---->
      </li>
    </ul>
  </div>
  <!---->
  <!---->
  <!---->
</div>"
`;

exports[`ComponentPanel when techDebts is set renders the techDebts 1`] = `
"<div>
  <h3 class=\\"text-info\\">Test Name</h3> <a href=\\"/components/test-id\\" target=\\"_self\\" class=\\"btn my-1 btn-info\\">
    More Info
  </a>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <div class=\\"my-4\\">
    <h5 class=\\"text-info\\">Tech Debts</h5>
    <ul>
      <li><span><span class=\\"badge badge-success\\">Missing Priority</span></span>
        <div>
          <p>Test Description 1</p>
        </div>
        <div></div>
        <!---->
      </li>
      <li><span><span class=\\"badge badge-success\\">Missing Priority</span></span>
        <div>
          <p>Test Description 2</p>
        </div>
        <div></div>
        <!---->
      </li>
    </ul>
  </div>
</div>"
`;
